Lib/test/test_asyncio/test_proactor_events.py

```diff
@@ -346,14 +346,31 @@ def test_pause_resume_reading(self):
         for i in range(10):
             self.loop._run_once()
         self.protocol.data_received.assert_called_with(b'data2')
+
+        # pause_reading is idepotent
+        tr.pause_reading()
+        tr.pause_reading()
+        self.assertTrue(tr._paused)
+
         tr.resume_reading()
         self.assertFalse(tr._paused)
         self.loop._run_once()
         self.protocol.data_received.assert_called_with(b'data3')
         self.loop._run_once()
         self.protocol.data_received.assert_called_with(b'data4')
+
+        # resume_reading is idepotent
+        tr.resume_reading()
+        tr.resume_reading()
+
         tr.close()
 
+        # pause/resume reading is no-op on closed transport
+        tr.pause_reading()
+        self.assertFalse(tr._paused)
+        tr._paused = True
```

Comment:
aiohttp maintains `paused` state separately from asyncio transport. it would be useful to have public api for that.

in aiohttp `paused` state is maintained by different edges, protocol pauses transport
when it get enough data, but resume happen from other edge, when user reads-out data from incoming stream. that is the reason why I think pause/resume should be ideponent and no-op during closing stage.